### PR TITLE
[RR-181] clear sessions on NO_OP

### PR DESCRIPTION
**Pull request by sjpotter**
Upon apply of a NO_OP, clear all active sessions.

### CODE REVIEW

File: src/raft.c [hunk 1]
```diff
@@ -575,6 +575,19 @@ static void handleEndClientSession(RedisRaftCtx *rr, raft_entry_t *entry)
...
+{
+    void *client_session;
+    RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(rr->client_session_dict, "^", NULL, 0);
+
+    while (RedisModule_DictNextC(iter, NULL, (void **) &client_session) != NULL) {
+        freeClientSession(client_session);
+    }
+    RedisModule_DictIteratorStop(iter);
+    RedisModule_FreeDict(rr->ctx, rr->client_session_dict);
+    rr->client_session_dict = RedisModule_CreateDict(rr->ctx);
```
Comment (fadidahanna) [inline src/raft.c:588]: We don't have to free and alloc the whole dictionary here, but we should make sure that we free it at `RedisRaftCtxClear()` (+ please make sure that we actually free there the inner objs memory, for other dicts as well).

File: src/redisraft.c [hunk 2]
```diff
@@ -2031,7 +2031,7 @@ RRStatus RedisRaftCtxInit(RedisRaftCtx *rr, RedisModuleCtx *ctx)
     rr->locked_keys = RedisModule_CreateDict(rr->ctx);
 
     /* acl -> user dictionary */
-    rr->acl_dict = RedisModule_CreateDict(ctx);
+    rr->acl_dict = RedisModule_CreateDict(rr->ctx);
 
     rr->client_session_dict = RedisModule_CreateDict(ctx);
```
[the remark this answers is not in the document]
Comment (sjpotter) [reply src/redisraft.c:2036]: I think I made this change in the other PR.

File: src/raft.c (hunk 1 above)
Comment (sjpotter) [reply src/raft.c:588]: So, I wasn't sure if we could delete items while iterating without breaking the iteration.

And, I wasn't sure if its actually less performance to do many individual deletes from the dict, vs just deleting and recreating the dict.

that's why I went this way, if we think its better in other way, that's ok too.

Comment (fadidahanna) [reply src/raft.c:588]: k, checked now on Redis - indeed it seems like your practice is the right one, so let's keep it.

**Review by fadidahanna — APPROVED** [review]

**Pull request merged**